plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2733,6 +2743,27 @@ private static Optional<String> find(Map<String, ColumnHandle> assignments, Proj
         return Optional.empty();
     }
 
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
```

Comment:
This change affects DML statements (e.g. DELETE) in addition to SELECT. Please add test cases about write operations.

Reply:
MERGE statement is still missing.